client/state/data-layer/wpcom-http/utils.js

```diff
@@ -70,13 +70,17 @@ export const getRequestKey = fullAction => {
 export const requestsReducerItem = (
 	state = null,
 	{ meta: { dataLayer: { lastUpdated, pendingSince, status } = {} } = {} }
-) =>
-	Object.assign(
+) => {
+	if ( status === undefined ) {
```

Comment:
@dmsnell not sure why we get garbage values when using a custom `meta.dataLayer.requestKey` I had to add this guard. We get one call after pending is set, where meta is empty.

Reply:
Didn't spot it near EoD but here's what the bad action looks like:

<img width="677" alt="Screen Shot 2019-12-04 at 5 50 56 PM" src="https://user-images.githubusercontent.com/1270189/70196816-bd13a500-16be-11ea-837d-69a9abc6d89b.png">

Reply:
cc @Automattic/team-calypso noting weird behavior here, but I'll try to tidy in a follow up. I think a temporary guard here is okay, since the reducer doesn't ever expect a status of undefined except on init.